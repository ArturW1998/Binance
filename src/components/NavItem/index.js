import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { LISTS } from '../../constants';

const { PAIRS_LIST } = LISTS;
const BTC = PAIRS_LIST[1];

const NavItem = ({ onClick, pair, market, className }) => {
  const classes = cn('nav-link', className, { active: market === pair });

  return (
    <li className="nav-item">
      <button className={classes} onClick={onClick} data-tab={pair}>
        {pair}
        <span className="d-none d-sm-inline"> Markets</span>
      </button>
    </li>
  );
};

NavItem.propTypes = {
  onClick: PropTypes.func,
  pair: PropTypes.string.isRequired,
  market: PropTypes.string,
  className: PropTypes.string,
};

NavItem.defaultProps = {
  onClick: () => {},
  market: BTC,
  className: '',
};

export default NavItem;
