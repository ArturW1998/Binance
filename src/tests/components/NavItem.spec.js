import React from 'react';
import { shallow } from 'enzyme';
import NavItem from '../../components/NavItem';

describe('NavItem component', () => {
  describe('NavItem component active prop', () => {
    describe('NavItem component with positive active', () => {
      const props = {
        active: true,
      };

      const navItem = shallow(<NavItem {...props} />);

      it('renders properly', () => {
        expect(navItem).toMatchSnapshot();
      });

      it('renders active', () => {
        expect(navItem.find('button').prop('className')).toEqual(
          'nav-link active',
        );
      });
    });

    describe('NavItem component with negative active', () => {
      const props = {
        active: false,
      };

      const navItem = shallow(<NavItem {...props} />);

      it('renders properly', () => {
        expect(navItem).toMatchSnapshot();
      });

      it('renders nav-link', () => {
        expect(navItem.find('button').prop('className')).toEqual('nav-link');
      });
    });
  });
  //
  describe('NavItem component onClick prop', () => {
    const mockOnClick = jest.fn();

    const props = {
      onClick: mockOnClick,
    };

    const navItem = shallow(<NavItem {...props} />);

    it('renders properly', () => {
      expect(navItem).toMatchSnapshot();
    });

    describe('when clicking the button', () => {
      beforeEach(() => {
        navItem.find('button').simulate('click');
      });

      it('dispatches the `onClick()` method it receives from props', () => {
        expect(mockOnClick).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe('NavItem component with pair', () => {
    const props = {
      pair: 'BTC',
    };

    const navItem = shallow(<NavItem {...props} />);

    it('renders properly', () => {
      expect(navItem).toMatchSnapshot();
    });

    it('renders pair', () => {
      expect(navItem.find('button').prop('data-tab')).toEqual(props.pair);
    });

    it('renders pair text', () => {
      expect(navItem.find('button').text()).toEqual(`${props.pair} Markets`);
    });
  });

  describe('NavItem component with className', () => {
    const props = {
      className: 'btn',
    };

    const navItem = shallow(<NavItem {...props} />);

    it('renders properly', () => {
      expect(navItem).toMatchSnapshot();
    });

    it('renders className', () => {
      expect(navItem.find('button').prop('className')).toEqual(
        `nav-link ${props.className}`,
      );
    });
  });

  describe('NavItem component with default props', () => {
    const mockOnClick = jest.fn();
    const props = {};

    const navItem = shallow(<NavItem {...props} />);

    it('renders properly', () => {
      expect(navItem).toMatchSnapshot();
    });

    describe('NavItem component pair prop', () => {
      it('renders default navigation item pair', () => {
        expect(navItem.find('button').prop('data-tab')).toEqual('BTC');
      });

      it('renders default navigation item pair text', () => {
        expect(navItem.find('button').text()).toEqual('BTC Markets');
      });
    });

    it('renders default button className', () => {
      expect(navItem.find('button').prop('className')).toEqual('nav-link');
    });

    describe('NavItem component onClick prop', () => {
      beforeEach(() => {
        navItem.find('button').simulate('click');
      });

      it('dispatches the `onClick()` method it receives from props', () => {
        expect(mockOnClick).toHaveBeenCalledTimes(0);
      });
    });
  });
});
