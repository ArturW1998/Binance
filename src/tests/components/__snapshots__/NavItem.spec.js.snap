// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem component NavItem component active prop NavItem component with negative active renders properly 1`] = `"<li class=\\"nav-item\\"><button class=\\"nav-link\\" data-tab=\\"BTC\\">BTC<span class=\\"d-none d-sm-inline\\"> Markets</span></button></li>"`;

exports[`NavItem component NavItem component active prop NavItem component with positive active renders properly 1`] = `"<li class=\\"nav-item\\"><button class=\\"nav-link active\\" data-tab=\\"BTC\\">BTC<span class=\\"d-none d-sm-inline\\"> Markets</span></button></li>"`;

exports[`NavItem component NavItem component onClick prop renders properly 1`] = `"<li class=\\"nav-item\\"><button class=\\"nav-link\\" data-tab=\\"BTC\\">BTC<span class=\\"d-none d-sm-inline\\"> Markets</span></button></li>"`;

exports[`NavItem component NavItem component with className renders properly 1`] = `"<li class=\\"nav-item\\"><button class=\\"nav-link btn\\" data-tab=\\"BTC\\">BTC<span class=\\"d-none d-sm-inline\\"> Markets</span></button></li>"`;

exports[`NavItem component NavItem component with default props renders properly 1`] = `"<li class=\\"nav-item\\"><button class=\\"nav-link\\" data-tab=\\"BTC\\">BTC<span class=\\"d-none d-sm-inline\\"> Markets</span></button></li>"`;

exports[`NavItem component NavItem component with pair renders properly 1`] = `"<li class=\\"nav-item\\"><button class=\\"nav-link\\" data-tab=\\"BTC\\">BTC<span class=\\"d-none d-sm-inline\\"> Markets</span></button></li>"`;
